File: Module 4/assignment code/Task 1 Pascal's triangle.py
def generate_pascals_triangle(num_rows):
    # Initialize a 2D list to store the values of the triangle
    triangle = [[0] * num_rows for _ in range(num_rows)]
    
    # Loop through each row
    for row in range(num_rows):
        # Loop through each column in the current row
        for col in range(row + 1):
            # The first and last elements in each row are 1
            if col == 0 or col == row:
                triangle[row][col] = 1
            else:
                # Other elements are the sum of the two elements above them
                triangle[row][col] = triangle[row - 1][col - 1] + triangle[row - 1][col]
            
            # Print the value at the current position
            print(triangle[row][col], end=" ")
        # Print a newline after each row
        print()

# Input number of rows from the user
num_rows = int(input("Enter number of rows: "))

# Generate and print Pascal's Triangle
generate_pascals_triangle(num_rows)

File: Module 4/assignment code/Task 1 Pascal's triangle.js
function generatePascalsTriangle(numRows) {
    // Create a 2D array to store the triangle values
    let triangle = Array.from({ length: numRows }, () => Array(numRows).fill(0));

    // Loop through each row
    for (let row = 0; row < numRows; row++) {
        // Loop through each column in the current row
        for (let col = 0; col <= row; col++) {
            // The first and last elements in each row are 1
            if (col === 0 || col === row) {
                triangle[row][col] = 1;
            } else {
                // Other elements are the sum of the two elements above them
                triangle[row][col] = triangle[row - 1][col - 1] + triangle[row - 1][col];
            }
            
            // Print the value at the current position
            process.stdout.write(triangle[row][col] + " ");
        }
        // Print a newline after each row
        console.log();
    }
}

// Input number of rows from the user
let numRows = parseInt(prompt("Enter number of rows: "), 10);

// Generate and print Pascal's Triangle
generatePascalsTriangle(numRows);
